src/piplapis/models/fields/vehicle.php

```diff
@@ -0,0 +1,160 @@
+<?php
+require_once dirname(__FILE__) . '/../utils.php';
+require_once dirname(__FILE__) . '/field.php';
+
+class PiplApi_Vehicle extends PiplApi_Field
+{
+    protected $children = array('vin', 'year', 'make', 'model', 'color', 'vehicle_type', 'display');
+    protected $attributes = array('is_vin_valid');
+
+    function __construct($params=array())
+    {
+        extract($params);
+        parent::__construct($params);
+
+        // vin
+        // year
+        // make
+        // model
+        // color
+        // vehicle_type
+        // is_vin_valid
+
+
+        if (!empty($vin))
+        {
+            $this->vin = $vin;
+        }
+        
+        if (!empty($year))
+        {
+            
+            $this->year = $year;
+        }
+        
+        if (!empty($make))
+        {
+            $this->make = $make;
+        }
+        
+        if (!empty($model))
+        {
+            $this->model = $model;
+        }
+        
+        if (!empty($color))
+        {
+            $this->color = $color;
+        }
+        
+        if (!empty($vehicle_type))
+        {
+            $this->vehicle_type = $vehicle_type;
+        }
+        
+        if (!empty($is_vin_valid))
+        {
+            $this->is_vin_valid = $is_vin_valid;
+        }
+
+        $this->display = $this->get_display();
+    }
+
+    protected function get_display(){
+        $attr_list = array('year', 'make', 'model', 'vehicle_type', 'color');
+        $display = array();
+        
+        foreach ($attr_list as $attr){
+            if (empty($this->$attr)){
+                continue;
+            }
+            
+            $display[] = ucwords(strtolower($this->$attr));
+        }
+
+        if (!empty($display)){
+            $display[] = '-';
+        }
+
+
+        $display[] = 'VIN';
+
+        if (!empty($this->vin)){
+            $display[] = $this->vin;
+        }
+
+
+        return implode(' ', $display);
+    }
+
+
+    public static function validate_vin_checksum($vin){
```

Comment:
why this function is static? we can instantiate 'vehicle' with vin and then call this method without passing the vin (same goes for  'is valid')

Reply:
i did it in order to stay consistent with the Python SDK, those functions are static there as well

Reply:
:+1: